Part1/Proj1/ViewController.swift

```diff
@@ -7,13 +7,90 @@
 
 import UIKit
 
-class ViewController: UIViewController {
+func isIPhone() -> Bool {
+return UIDevice.current.userInterfaceIdiom == .phone
+}
+class someVC:
+UIViewController,UICollectionViewDelegate,UICollectionViewDelegateFlowLayout,UICollectionViewDataSource {
+@IBOutlet weak var collectionViewTopConstraint: NSLayoutConstraint!
+@IBOutlet var detailViewWidthConstraint: NSLayoutConstraint!
+@IBOutlet var collectionView: UICollectionView!
+@IBOutlet var detailView: UIView!
+var detailVC : UIViewController?
+var dataArray : [Any]?
+override func viewWillAppear(_ animated: Bool) {
+fetchData()
+self.collectionView.dataSource = self
+self.collectionView.delegate = self
+self.collectionView.register(UINib(nibName: "someCell",
+
+bundle: nil), forCellWithReuseIdentifier: "someCell")
 
-    override func viewDidLoad() {
-        super.viewDidLoad()
-        // Do any additional setup after loading the view.
-    }
+self.navigationItem.rightBarButtonItem = UIBarButtonItem.init(title:NSLocalizedString("Done", comment:
+""), style: .plain, target: self, action: #selector(dissmissController))
+}
+@objc func dissmissController () {}
+func fetchData() {
+let url = URL(string: "testreq")!
+let task = URLSession.shared.dataTask(with: url) { [self](data, response, error) in
+self.dataArray = data as? [Any]
+self.collectionView.reloadData()
+}
+task.resume()
+}
+@IBAction func closeshowDetails () {
```

Comment:
showDetail() and closeShowDetails() Functions:
   - These functions handle the showing and closing of a detail view. To enhance code clarity and maintainability, consider adding comments or documentation for these functions to explain their purpose and usage. Additionally, it might be beneficial to include some error handling or validation specific to these operations, such as checking if the `detailView` is already displayed or if the `detailVC` is non-null.
   - Use `[weak self]` in closures within these functions to prevent potential retain cycles and ensure safe memory management. For example:
     ```swift
     [weak self] in
     self?.view.layoutIf